### DISCUSSION TITLE
Release notes of npm package hosted in gitlab registry are not populated from CHANGELOG.md

### DISCUSSION THREAD

**Discussion by mzglinski** [renovatebot/renovate#14291] [Request Help]
I've created an example group to replicate my issue https://gitlab.com/renovate-internal-release-notes

To my understanding, all you need to do is reference the gitlab repository in the package.json file, which I've done [here](https://gitlab.com/renovate-internal-release-notes/dependency/-/blob/main/package.json#L6).

However, the [merge request](https://gitlab.com/renovate-internal-release-notes/application/-/merge_requests/2) does not contain information about the release notes.

[Renovate logs](https://gitlab.com/renovate-internal-release-notes/renovate-bot/-/jobs/2107882626)

Am I missing something or is this a bug?

**Comment by rarkins** [comment]
As a next step, can you query the gitlab registry for that package? e.g. I want the equivalent of `https://registry.npmjs.org/left-pad` but for the package you're referencing/updating. We need to check if GitLab includes the required metadata there.

**Comment by mzglinski** [comment]
Thanks for the reply! 

I believe these are the links you are asking for:

https://gitlab.com/api/v4/projects/33828436/packages/
https://gitlab.com/api/v4/groups/16180206/packages/
https://gitlab.com/api/v4/projects/33828436/packages/5214220

I've constructed those links based on this documentation: https://docs.gitlab.com/ee/api/packages.html

**Reply by rarkins** [reply]
Does it look like the source URL is included in the registry JSON?

**Reply by mzglinski** [reply]
Also, probably this one: https://gitlab.com/api/v4/packages/npm/@renovate-internal-release-notes%2fdependency I've checked and this is the URL that `npm view` uses.

> Does it look like the source URL is included in the registry JSON?

No, it does not, so GitLab Package Registry (for that functionality) is not supported by Renovate, is that what you are saying?

**Reply by rarkins** [reply]
Renovate supports GitLab package registry fine, but if GitLab's npm registry support is only partial and strips out the repository metadata about your packages then Renovate won't know the source URL to look up release notes from.

**Reply by rarkins** [reply]
I think this looks like the spec: https://github.com/npm/registry/blob/master/docs/REGISTRY-API.md

Note the `repository` field under both Package and Version, which seems to be missing from GitLab's implementation?

**Reply by mzglinski** [reply]
Yes, this seems to be the GitLab implementation https://docs.gitlab.com/ee/user/packages/npm_registry/#npm-metadata.

For now, they only support the [abbreviated form](https://github.com/npm/registry/blob/master/docs/responses/package-metadata.md#abbreviated-version-object), and the full form is still in the backlog https://gitlab.com/gitlab-org/gitlab/-/issues/344107

Anyway, thanks for the help, we need to wait for gitlab to implement the full form I guess.
